coalib/misc/Exceptions.py

```diff
@@ -3,6 +3,8 @@
 from coalib.misc import Constants
 from coalib.output.printers.LogPrinter import LogPrinter
 
+from pkg_resources import ContextualVersionConflict, VersionConflict
```

Comment:
This issue has irritated so much that I'd rather just put both "in case".

In side contextual isn't a subclass because I tried isinstance(ContextualVersionConflict, VersionConflict) and it was false.

Lets put both and ignore scrutinizer

-----Original Message-----
From: "Adhityaa Chandrasekar" <notifications@github.com>
Sent: ‎17-‎03-‎2016 17:27
To: "coala-analyzer/coala" <coala@noreply.github.com>
Cc: "AbdealiJK" <abdealikothari@gmail.com>
Subject: Re: [coala] Exceptions: Handle package version conflicts (#1894)

In coalib/misc/Exceptions.py:
> @@ -3,6 +3,8 @@
>  from coalib.misc import Constants
>  from coalib.output.printers.LogPrinter import LogPrinter
>  
> +from pkg_resources import ContextualVersionConflict, VersionConflict
I just experimented around with different stuff and I'm now fairly certain that VersionConflict is enough: I think ContextualVersionConflict is derived from VersionConflict and ContextualVersionConflict is raised sometimes in the newer versions, but VersionConflict is raised always in the older ones.
And when I put just VersionConflict with the newer version, it always capturing the exception. Go ahead with it?
—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub